### PR TITLE
missing: split missing.h into small pieces 

### PR DESCRIPTION
**Pull request by yuwata**
Continuation of recent work.

### CODE REVIEW

File: src/basic/missing_input.h [hunk 1]
```diff
@@ -11,9 +12,9 @@
 /* linux@06a16293f71927f756dcf37558a79c0b05a91641 (4.4) */
 #ifndef EVIOCSMASK
 struct input_mask {
-        uint32_t type;
-        uint32_t codes_size;
-        uint64_t codes_ptr;
+        __u32 type;
+        __u32 codes_size;
+        __u64 codes_ptr;
```
Comment (keszybz) [inline src/basic/missing_input.h:17]: What is the advantage of this?

File: src/basic/missing.h [hunk 2]
```diff
@@ -87,14 +86,6 @@ struct sockaddr_vm {
...
 #endif
 
-#ifndef OOM_SCORE_ADJ_MIN
-#define OOM_SCORE_ADJ_MIN (-1000)
-#endif
-
-#ifndef OOM_SCORE_ADJ_MAX
-#define OOM_SCORE_ADJ_MAX 1000
-#endif
-
```
Comment (keszybz) [inline src/basic/missing.h:98]: Isn't it used in parse-util.c and process-util.c?

File: src/basic/missing.h [hunk 3]
```diff
@@ -60,14 +60,6 @@
 #  define CIFS_MAGIC_NUMBER 0xFF534D42
 #endif
 
-#ifndef DRM_IOCTL_SET_MASTER
-#  define DRM_IOCTL_SET_MASTER _IO('d', 0x1e)
-#endif
-
-#ifndef DRM_IOCTL_DROP_MASTER
-#  define DRM_IOCTL_DROP_MASTER _IO('d', 0x1f)
-#endif
```
Comment (keszybz) [inline src/basic/missing.h:225]: Isn't this still used in logind?

File: src/basic/missing.h [hunk 4]
```diff
@@ -68,14 +67,6 @@
 #define NS_GET_NSTYPE _IO(0xb7, 0x3)
 #endif
 
-#ifndef FALLOC_FL_KEEP_SIZE
-#define FALLOC_FL_KEEP_SIZE 0x01
-#endif
-
-#ifndef FALLOC_FL_PUNCH_HOLE
-#define FALLOC_FL_PUNCH_HOLE 0x02
```
Comment (keszybz) [inline src/basic/missing.h:305]: Isn't this still used in fs-util.c?

File: src/basic/missing_input.h (hunk 1 above)
Comment (yuwata) [reply src/basic/missing_input.h:17]: I guess this has no big advantage. I just want to keep the current kernel definitions. Also, by using this way, we do not need any additional headers to implement each `missing_*.h` except `linux/types.h`.

File: src/basic/missing.h (hunk 2 above)
Comment (yuwata) [reply src/basic/missing.h:98]: These are defined in linux/oom.h and the header is already included.

File: src/basic/missing.h (hunk 3 above)
Comment (yuwata) [reply src/basic/missing.h:225]: These are provided by kernel earlier than 3.11 in drm/drm.h. So, logind now includes drm/drm.h.

File: src/basic/missing.h (hunk 4 above)
Comment (yuwata) [reply src/basic/missing.h:305]: Similar to the others, which has been provided earlier than kernel-3.11.

**Comment by yuwata** [comment]
So, should I update commit messages after all CIs finish?

File: src/basic/missing_input.h (hunk 1 above)
Comment (keszybz) [reply src/basic/missing_input.h:17]: That last thing is certainly a valid point.

File: src/basic/missing.h (hunk 2 above)
Comment (keszybz) [reply src/basic/missing.h:98]: It was added in v2.6.35-5872-ga63d83f427. I guess that's old enough.

**Comment by yuwata** [comment]
BTW, I am planning the following:
- replace each included missing.h with relevant smaller missing_xxx.h.
- drop missing.h
- move several missing_xxx.h to shared/ if possible.
- rename shared/linux/*.h and shared/linux-3.13/*.h to missing_*.h

**Comment by yuwata** [comment]
@keszybz Forece-pushed new version. Only commit messages are updated.

**Comment by yuwata** [comment]
@mrc0mmand CentOS CI seems to have a problem now...

**Comment by mrc0mmand** [comment]
@yuwata yeah, that because of https://pagure.io/fedora-infrastructure/issue/7422 ...

**Comment by yuwata** [comment]
Ah, I see. Thanks.

**Pull request merged**